### ISSUE TITLE
OVH: OS_USER_DOMAIN_ID

### ISSUE THREAD

**Issue by rendaw** [gophercloud/gophercloud#2045]
OVH provides an rc file with these keys:
```
OS_AUTH_URL
OS_PROJECT_ID
OS_PROJECT_NAME
OS_USER_DOMAIN_NAME
OS_USER_DOMAIN_ID
OS_USERNAME
OS_PASSWORD
OS_REGION_NAME
OS_INTERFACE
OS_IDENTITY_API_VERSION
```

The Env auth fails due to missing `OS_DOMAIN_NAME` or `OS_DOMAIN_ID`.  Renaming `OS_USER_DOMAIN_NAME` seemed to work for me but my knowledge of OS auth is very limited.  It looks like perhaps the `OS_USER_DOMAIN_*` keys were used before but not any more?

From a UX perspective it would be nice if OVH rc files worked as they provide them, given how much of a black hole auth is.

**Comment by kayrus** [comment]
@rendaw Indeed this repo has some logic issues. I recommend to use the https://pkg.go.dev/github.com/gophercloud/utils/openstack/clientconfig package to handle the auth parameters.

**Comment by rendaw** [comment]
Oh perfect, thanks!  Would it make sense to recommend this in the main repo readme?

**Comment by jtopjian** [comment]
Yeah, I've been meaning to do that for a while. I'll leave this issue open as a reminder - hopefully I'll get to it within a week.

> Indeed this repo has some logic issues.

To clarify: Nowadays, this repo has "just enough" logic to get acceptance tests to work, otherwise any environment variable support would have been removed entirely. And that should certainly be documented.

**Comment by jtopjian** [comment]
I've opened #2053 which copies the note found [here](https://github.com/gophercloud/gophercloud/blob/master/doc.go#L12-L17) to the main README.md file.

**Comment by jtopjian** [comment]
Should be resolved in #2053
